app/components/Button.tsx

```diff
@@ -4,9 +4,8 @@ interface ButtonProps {
   size: 'small' | 'medium' | 'large';
   label: string;
   varient: 'primary' | 'secondary';
-
   disabled?: boolean;
-  handleClick?: (value: React.SetStateAction<number>) => void;
+  handleClick?: () => void;
```

Comment:
should we add an action type except number instead of keeping it empty?

Reply:
> should we add an action type except number instead of keeping it empty?

`handleClick` was expecting a param earlier and since this is a button we are not passing any param. Do we want to catch the event? as it gives `React.MouseEvent<HTMLButtonElement, MouseEvent>` by default.

Reply:
Button Component should be. able to access any data type if and when used, restricting the type to number might create issue in future, @iRohitGaur , to ensure the types gets passed, is it handled in appropriate components where we are using Button

Reply:
> Button Component should be. able to access any data type if and when used, restricting the type to number might create issue in future, @iRohitGaur , to ensure the types gets passed, is it handled in appropriate components where we are using Button

agreed! That is one of the reasons I removed the existing type which was restricting it to number.

addressing your question, that `Button` component is only used in `contentPill` and there `handleClick` is not even used at this point as its optional.

This component had nothing to do with my code but I had to change it as its test was failing with the error: `Type 'MouseEvent<HTMLButtonElement, MouseEvent>' is not assignable to type 'SetStateAction<number>'` which is true. I'm not sure how this got skipped in previous PR but I thought of fixing it here.